cumulusci/core/tests/test_utils_merge_config.py

```diff
@@ -260,3 +262,15 @@ def test_cleanup_flow_step_override_conflicts__obsolete_flow_with_tasks(
     clean_configs = utils.cleanup_flow_step_override_conflicts(configs)
 
     assert expected_project_config == clean_configs["project_config"]
+
+
+def test_link_missing_task_or_flow():
```

Comment:
I'm not really a fan of this kind of testing. I think the automated tests should do what you described in the PR cmments. As a maintainer I don't care much that the new function `link_missing_task_or_flow` does the right thing. I care that CCI configs merge properly. I'd like to see some CCI configs merge. Then if I change the underlying implementation later, I'll still know that CCI configs merge properly.

For example, here is how we test the CCI validating linter.

https://github.com/SFDO-Tooling/CumulusCI/blob/8d46f48da6850d4b979d7f36f01f4b20103ed236/cumulusci/utils/yaml/tests/test_cumulusci_yml.py#L97

Reply:
IMO: A good rule of thumb is whether there's a clear line between the WI ACs and the test cases.

Reply:
I agree that we should build new tests to catch this type of problem, and I will write a WI to do so (along with documenting this specific wrinkle of flow customization). However, I don't want to hold up @Br4nd0R's fix to this bug to do that, as there is serious production impact we need to cure.